### ISSUE TITLE
Feature Request: Digital Signatures for APIs

### ISSUE THREAD

**Issue by derjones** [hendt/ebay-api#132]
Hey,

starting at 31.01.23 digital signatures are required for specific API calls in EU and UK. (https://developer.ebay.com/develop/guides/digital-signatures-for-apis) The keyManagement API calls are already implemented, but the signatures headers are not added to the specified API calls. Also a createSigningKey request and persisting the keys to a file would be nice.

Ebay created a lib to generate this signatures: https://github.com/ebay/digital-signature-nodejs-sdk

I just saw that you are working on a RC for 8.0.0, do you have plans to add this to this version?

Thanks Jonas

**Comment by dantio** [comment]
The RC does not include the digital signatures. 
I'll definitely work on this feature and hope to release it this year.
Feel free to contribute!

**Comment by derjones** [comment]
Okay, sure. But lets wait till the version 8 is released to avoid conflicts.

**Comment by dantio** [comment]
@derjones 
I'll release version 8 this probably this week after testing it under production env.
The digital signature feature is already in review. I already had successfully calls with finance API and GetAccount.

**Comment by derjones** [comment]
@dantio Thats great news, thank you for implementing this feature!

**Comment by dantio** [comment]
@derjones 
Might you like to test the digital signature? 
`npm i ebay-api@8.2.0-RC.0`

```js
let eBay = new eBayApi({ 
  appId: '...',
  certId: '...'
});
const signingKey = await eBay.developer.keyManagement.createSigningKey('ED25519');
// save the JSON somewhere

eBay = new eBayApi({
  appId: '...',
  certId: '...',
  signature: {
    jwe: signingKey.jwe,
    privateKey: signingKey.privateKey
  }
});

const summary = await eBay.sell.finances.sign.getSellerFundsSummary();
```

**Comment by derjones** [comment]
Yes! Sry that it took so long...
I just testet to create a digital signature and do a request like you suggested, works like a charm.
I also tested:
- Issue refund (Fulfillment API)
- Issuse refund (Post Order API)
- Proess return request (Post Order API)

Thank you for implementing this features!

**Comment by dantio** [comment]
@derjones 
Thank you! I'll merge the feature and release it soon.

**Comment by dantio** [comment]
Released in `8.4.0`
